modules/openapi-generator/src/main/resources/swift5/api.mustache

```diff
@@ -190,6 +199,7 @@ extension {{projectName}}API {
     {{/isDeprecated}}
     @available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}open{{/nonPublicApi}} class func {{operationId}}({{#allParams}}{{paramName}}: {{#isEnum}}{{#isContainer}}{{{dataType}}}{{/isContainer}}{{^isContainer}}{{{datatypeWithEnum}}}_{{operationId}}{{/isContainer}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}{{^required}}? = nil{{/required}}{{^-last}}, {{/-last}}{{/allParams}}{{#hasParams}}, {{/hasParams}}apiResponseQueue: DispatchQueue = {{projectName}}API.apiResponseQueue) async throws{{#returnType}} -> {{{returnType}}}{{/returnType}} {
+
         return try await withCheckedThrowingContinuation { continuation in
```

Comment:
@DavidHorvath  Can you please try this?
```
    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}open{{/nonPublicApi}} class func {{operationId}}({{#allParams}}{{paramName}}: {{#isEnum}}{{#isContainer}}{{{dataType}}}{{/isContainer}}{{^isContainer}}{{{datatypeWithEnum}}}_{{operationId}}{{/isContainer}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}{{^required}}? = nil{{/required}}{{^-last}}, {{/-last}}{{/allParams}}{{#hasParams}}, {{/hasParams}}apiResponseQueue: DispatchQueue = {{projectName}}API.apiResponseQueue) async throws{{#returnType}} -> {{{returnType}}}{{/returnType}} {
        var task: URLSessionTask?
        return try await withTaskCancellationHandler {
            try await withCheckedThrowingContinuation { continuation in
                task = {{operationId}}WithRequestBuilder({{#allParams}}{{paramName}}: {{paramName}}{{^-last}}, {{/-last}}{{/allParams}}).execute(apiResponseQueue) { result in
                    switch result {
                    {{#returnType}}
                    case let .success(response):
                        continuation.resume(returning: response.body!)
                    {{/returnType}}
                    {{^returnType}}
                    case .success:
                        continuation.resume(returning: ())
                    {{/returnType}}
                    case let .failure(error):
                        continuation.resume(throwing: error)
                    }
                }
            }
        } onCancel: { [task] in
            task?.cancel()
        }
    }
```

Reply:
I hope I got it right, because I didn't tested it, but I tested the swift code that I send I minute ago

Reply:
It wasn't very straightforward at first how to do it but finally I was able to test our generated async await APIClient.
I've displayed the response on the UI or a `Cancelled` text if the task had been canceled.

``` swift
    func load() async throws -> String {
        let fetchTask = Task { () -> String in
            do {
                let response = try await TestAPI.asyncTest()
                if Task.isCancelled { return "Cancelled" }

                return response
            } catch {
                return "Error"
            }
        }

        fetchTask.cancel()

        return await fetchTask.value
    }

    open class func asyncTest(apiResponseQueue: DispatchQueue = TestAPI.apiResponseQueue) async throws -> String {
        var task: URLSessionTask?
        return try await withTaskCancellationHandler {
            try await withCheckedThrowingContinuation { continuation in
                task = testWithRequestBuilder().execute(apiResponseQueue) { result in
                    switch result {
                    case let .success(response):
                        continuation.resume(returning: response.body!)
                    case let .failure(error):
                        continuation.resume(throwing: error)
                    }
                }
            }
        } onCancel: { [task] in
            print("Task Cancelled")
            task?.cancel()
        }
    }

```

And there was a `Task Cancelled` console message and a `Cancelled` text on the UI as well, so it works! 🙂 🚀

Reply:
Nice, good to know 👍